worlds/sc2wol/MissionTables.py

```diff
@@ -408,7 +503,8 @@ class SC2CampaignGoal(NamedTuple):
 
 campaign_final_mission_locations: Dict[SC2Campaign, SC2CampaignGoal] = {
     SC2Campaign.WOL: SC2CampaignGoal(SC2Mission.ALL_IN, "All-In: Victory"),
-    SC2Campaign.PROPHECY: SC2CampaignGoal(SC2Mission.IN_UTTER_DARKNESS, "In Utter Darkness: Kills")
+    SC2Campaign.PROPHECY: SC2CampaignGoal(SC2Mission.IN_UTTER_DARKNESS, "In Utter Darkness: Kills"),
+    SC2Campaign.HOTS: SC2CampaignGoal(SC2Mission.THE_RECKONING, "The Reckoning: Victory")
```

Comment:
The Reckoning in shuffled campaign already works. As far as I understand, the generator just treats it like any other Hard mission so long as HotS isn't the hardest campaign available (so if WoL is included).

See Dominion Space here (WoL/Prophecy not pictured):
![python_PklzdoZuaN](https://github.com/Ziktofel/Archipelago/assets/30208051/bd9da6b8-0866-45a6-b42d-4ce4469aa882)

Reply:
The thing is that if we want to randomize The Reckoning in a shuffled campaign by default if we don't have any mission excluded

Reply:
Right. HotS had an option to choose between The Reckoning and some hard missions, but that didn't really easily translate into how the cross-campaign code worked.

I'm changing The Reckoning to be an alt goal, but the rest of the suggestion only works for Vanilla order because all the other places already consider `None` as a possible primary goal, so I believe it won't be possible for now to pick a non-Vanilla order and guarantee The Reckoning as the goal without excluding a bunch of other missions. This could be an option later, but for now I think it's fine.

Reply:
I think that the default HotS goal should be random. Reckoning is used for Vanilla. the None in primary goal is to be able to have a random goal by default

Reply:
Ok, that's what the behavior is now.